packages/protocol/scripts/truffle/make-release.ts

```diff
@@ -143,43 +141,25 @@ const deployProxy = async (
   const Proxy = await artifacts.require(`${contractName}Proxy`)
   // Hack to trick truffle, which checks that the provided address has code
   const proxy = await (dryRun ? Proxy.at(REGISTRY_ADDRESS) : Proxy.new())
+
   const initializeAbi = (contract as any).abi.find(
     (abi: any) => abi.type === 'function' && abi.name === 'initialize'
   )
-  console.log(`Setting ${contractName}Proxy implementation to ${contract.address}`)
-  if (initializeAbi) {
-    const args = initializationData[contractName]
-    console.log(`Initializing ${contractName} with: ${args}`)
-    if (!dryRun) {
-      await setAndInitializeImplementation(
-        web3,
-        proxy,
-        contract.address,
-        initializeAbi,
-        {},
-        ...args
-      )
-    }
-  } else {
+  // Only set implementation if there is no initialize on the implementation contract,
+  // as otherwise anyone could initialize via the proxy and become the owner
+  if (!initializeAbi) {
```

Comment:
I would suggest never doing this during the deployment, always setting implementation in governance proposal. That way the behavior is always the same (for every registry repoint in a proposal, there is an implementation setter as well), and will make the storage of each fresh Proxy exactly the same, making verifying this fact easier.